backup-stores/azure/src/main/java/io/camunda/zeebe/backup/azure/ManifestManager.java

```diff
@@ -0,0 +1,194 @@
+ */
+package io.camunda.zeebe.backup.azure;
+
+import static com.fasterxml.jackson.databind.SerializationFeature.WRITE_DATES_AS_TIMESTAMPS;
+
+import com.azure.core.http.rest.Response;
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.Context;
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.models.BlobErrorCode;
+import com.azure.storage.blob.models.BlobRequestConditions;
+import com.azure.storage.blob.models.BlobStorageException;
+import com.azure.storage.blob.models.BlockBlobItem;
+import com.azure.storage.blob.options.BlobParallelUploadOptions;
+import com.azure.storage.common.implementation.Constants;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import io.camunda.zeebe.backup.api.Backup;
+import io.camunda.zeebe.backup.api.BackupIdentifier;
+import io.camunda.zeebe.backup.azure.AzureBackupStoreException.UnexpectedManifestState;
+import io.camunda.zeebe.backup.azure.manifest.Manifest;
+import io.camunda.zeebe.backup.azure.manifest.Manifest.InProgressManifest;
+import io.camunda.zeebe.backup.azure.manifest.Manifest.StatusCode;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+
+public final class ManifestManager {
+  public static final int PRECONDITION_FAILED = 412;
+
+  /**
+   * The path format consists of the following elements:
+   *
+   * <ul>
+   *   <li>{@code "manifests"}
+   *   <li>{@code partitionId}
+   *   <li>{@code checkpointId}
+   *   <li>{@code nodeId}
+   *   <li>{@code "manifest.json"}
+   * </ul>
+   *
+   * The path format is constructed by partitionId/checkpointId/nodeId/manifest.json
+   */
+  private static final String MANIFEST_PATH_FORMAT = "%s/%s/%s/manifest.json";
+
+  private static final ObjectMapper MAPPER =
+      new ObjectMapper()
+          .registerModule(new Jdk8Module())
+          .registerModule(new JavaTimeModule())
+          .disable(WRITE_DATES_AS_TIMESTAMPS)
+          .setSerializationInclusion(Include.NON_ABSENT);
+  private boolean containerCreated = false;
+  private final BlobContainerClient blobContainerClient;
+
+  ManifestManager(final BlobContainerClient blobContainerClient) {
+    this.blobContainerClient = blobContainerClient;
+  }
+
+  PersistedManifest createInitialManifest(final Backup backup) {
+
+    final var manifest = Manifest.createInProgress(backup);
+    final byte[] serializedManifest;
+    assureContainerCreated();
+    try {
+      serializedManifest = MAPPER.writeValueAsBytes(manifest);
+    } catch (final JsonProcessingException e) {
+      throw new RuntimeException(e);
+    }
+    final BlobClient blobClient = blobContainerClient.getBlobClient(manifestPath((manifest)));
```

Comment:
:upside_down_face: Double parenthesis 
```suggestion
    final BlobClient blobClient = blobContainerClient.getBlobClient(manifestPath(manifest));
```